csharp-intermediate-topics/RenameFilesWithCSharp/RenameFilesWithCSharp/BatchRenameFileWithErrorHandling.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace RenameFilesWithCSharp;
+
+public class BatchRenameFileWithErrorHandling
+{
+    public static void RenamePhotosWithDateTime(string directoryPath)
+    {
+        try
+        {
+
+            string[] allowedExtensions = { ".jpg", ".png", ".jpeg", ".gif" };
+
+            var photoFiles = Directory.GetFiles(directoryPath)
```

Comment:
We should change this to Directory.EnumerateFiles as that returns an IEnumerable rather than an array.
We could honestly even remove the ToList call since we are just going to iterate over this in a foreach loop